### ISSUE TITLE
RFC: for people in buildings

### ISSUE THREAD

**Issue by Charlotte-Morgan** [inasafe/inasafe#1179]
# Problem

InaSAFE impact functions assessing the impact of hazards on people rely on raster population data. There are limited sources of raster population data (Landscan 2008, AsiaPop / WorldPop) and many InaSAFE users have reported that the analysis results based on these population data are inaccurate. 

This RFC explores one idea for improving the accuracy of affected population results that InaSAFE produces for impacts on people.
# Duration

November - December 2014
# Current State

This proposal is in **draft - please contribute**.
# Proposers
- Charlotte Morgan (@Charlotte-Morgan)
- Christian Christelis (christian@kartoza.com @cchristelis)
# Detail

InaSAFE impact functions assessing the impact of hazards on people rely on raster population data. There are limited sources of raster population data (Landscan 2008, AsiaPop / WorldPop) and many InaSAFE users have reported that the analysis results based on these population data are inaccurate.  Indonesian users are keen to be able to use Indonesian census data to determine affected populations. At present Indonesian census data (as in many parts of the world) are captured and distributed as polygon data. The challenge is how to turn these census data into a meaningful dataset that can be used in InaSAFE. & to improve the assessment of affected populations.

There are at least four options:
1. Rasterise the census data without any expert input regarding population distribution within the census collection area.
2. Supply high resolution census population data to population modelling experts and get them to generate raster data suitable for use in InaSAFE.
3. Make InaSAFE run on vector population data.
4. In areas where buildings have been mapped, statistically assign the population to buildings.

This RFC seeks to explore option 4 above in more detail and includes two issues:
1. How to generate a new combined exposure data set from existing population and buildings data. This is the focus of this ticket but should not forget how the data will be used in IF in the future.
1. The need for a new IF to assess impact on people when the exposure data consists of people in buildings. New IF should also consider the capacity to include both building and population vulnerability attributes if and when these are available some time in the future. This component should be expanded as a separate ticket.
# Proposed solution

That we limit consideration for this new exposure data set to the combination of OSM buildings and census population data at the highest capture resolution (smallest block sizes).
That we assume the census records 'usual place of residence' and as such is where people live / sleep.

OSM buildings will need a certain number of attributes to be populated (filled in) in order to support this proposal. OSM buildings will need to be classified by type (those used for residential purposes) and include attributes for size (footprint and number of floors used for residential purposes).

That the population could be statistically distributed between the buildings based on building floor area.
eg 
![image](https://cloud.githubusercontent.com/assets/5743135/5102902/e679e934-7003-11e4-8da1-1dc0dabc166a.png)

Requirements to achieve this:
all buildings with in the census collection area are mapped, with core attributes of footprint and floors.
the census collection boundary is aligned with the mapped buildings.
# Suggestion

This idea be tested manually in a rural community or on a small island (Rokotenda) where there is a clear association between the census boundary and mapped land / buildings.
# vision for future

building attributes would include vulnerability - ie a measure of roof construction to determine susceptibility to ash fall loads, or cross bolts in roof frame for wind resistance etc. ie people in a well constructed house would not be affected by the same impact that would affect people in a kurang baik house.
population attributes such as gender and age proportions, economic status etc could also be assigned at the local level to identify vulnerable communities better than is possible with global proportions.
# Issues

day and night time populations are different and likely to be in different buildings.
expect building and population data to be misaligned and that this will need to be rectified prior to assigning people to buildings.
# See also
#488
# Record of votes
# Resolution
# CC

@Charlotte-Morgan 
@cchristelis
@mahardika 
@iyan31 
@maning 
@davidr007 
@timlinux 
@fredychandra 
@

**Comment by timlinux** [comment]
@Charlotte-Morgan note that the above link no longer works - can you update it with a current one?

**Comment by timlinux** [comment]
Note in the OSM downloader server side logic:

* Government
* **Residential**
* School
* Clinic/Doctor
* Place of Worship
* Hospital
* Commercial
* Industrial
* Utility

We propose that only residential type above gets used in calculations (as per @Charlotte-Morgan 's scheme above). Is that ok? Or should we include any of the above classes too?

**Comment by timlinux** [comment]
![peopleinbuildings - new page](https://cloud.githubusercontent.com/assets/178003/5939749/1347205c-a70f-11e4-8411-7c009bc391ab.png)

Above is an illustration to accompany an update we would like to give to @Charlotte-Morgan on proposed implementation approach.

**Comment by cchristelis** [comment]
There are a couple of concerns that I would like us to consider. Also @Charlotte-Morgan 's input will be valuable here.

# The Zones A, B, C

Census areas may be fully contained in the buildings. In this case the census area is made up of a single zone A as shown above. If the census area is only partially contained by buildings, then we have a zone B overlapped by buildings and zone C not building data. The final situation is census data with no building data resulting in only a zone C. 

In zone A we know the population count as well as all the buildings. Therefore we can make a good estimate of the people per building. So here we have a (high/good/90%/... help) confidence in what we have assigned. Rule 1

In zone B we know information about the buildings, but not enough about haw many people live in that area (zone C's affect cannot be ignored) to make a direct estimation based on the census data. Ignoring this zone, could lead with many x sections. X sections would lead to problems with analysis issues (having buildings in an analysis which does not contain any population => skewing). So to avoid this we have to estimate the number of people per building this could be done using averages or extrapolation. So here we have a (medium/moderate/50%... ) confidence in our estimate. Rule 2.

In zone C, nothing.

# Using a separate tool

If we integrate people per buildings estimation in the IF directly, we run the risk of ending up with no zone A. We would force the user to do an analysis on an area including at least not necessarily their points of interest. 

Furthermore we would make the best estimation using the largest possible extent to perform estimation. Hence it makes the most sense to have a estimate people in buildings tool, which makes an estimate, before running the IF. Also we could educate the user with this tool. 

This tool would expect:
1) census polygon data. 
2) building data with: number of floors and use 

# Pre-processing the people estimate

Since this is a slightly more i

**Comment by iyan31** [comment]
@timlinux I think it would be difficult to only use building=residential, since the number of building using this tags is really minimum. I just run quick calculation using osmosis, the number of building=residential is 23,000 in contrast with building=* where there are 1,9million for the whole Indonesia.

**Comment by timlinux** [comment]
Hi @iyan31 

We also use landuse %residential% in our filters see:

https://github.com/kartoza/osm-reporter/blob/develop/reporter/resources/buildings/transform.sql

Here are the relevant rules.

```sql

UPDATE planet_osm_polygon SET "type" = 'Residential' WHERE landuse  ILIKE
'%residential%' OR "building:use"='residential';

UPDATE planet_osm_polygon SET "type" = 'Residential' WHERE
  "building:use" = 'residential' AND "type" IS NULL ;

```

Can you suggest additional filters that will help us to identify residential dwellings?

**Comment by iyan31** [comment]
Hi @timlinux 

How about building=house ? I see more building use this compare to building=residential (23,000 vs 57,397) although it is still much much less than building=yes (1,3million).

**Comment by timlinux** [comment]
Hi @iyan31 

What about we use ``building=house`` as well, apply the other rules and then apply residential to anything else that is not tagged specifically as gov building, place of worship etc?

Regards

Tim

**Comment by Charlotte-Morgan** [comment]
@timlinux, @iyan31@ @cchristelis : Thanks for great thoughts.
- [ ] re building types. "residential" for the purposes of assigning people will be any building that people may live in regardless of its other main use. eg also a shop etc
The building-types will need to be classified according to local customs into those that get people assigned and those that don't. This should not overwrite the building type field used for post processing affected buildings in IF where buildings are the exposure data 

- [ ] People in buildings is effectively about generating a new type of exposure data. ie  we need to separate the generation of the data from the running of an IF. ie +1 for Rule 3 & +1 this a separate process requiring user guidance.
* i'm thinking that assigning people to buildings is more akin with existing post processing procedures than IF analyses.
* the pictures and rules above tend to have an IF analysis approach. I think Rule 1 is good but not sure about Rule 2.
* lets think of the census boundaries as an aggregation layer - the post processors currently run on a non square bounding box if an aggregation layer is defined.
* for the subsequent IF, maybe the analysis extent will also be constrained by the same census boundary?

So:
- [ ] Census area A1 will assign the population to buildings (identified as having a residential function or maybe inversely to everything other than the ones identified as non residential) within A1. Census area A2 will assign population to buildings within A2.etc .
- [ ] Buildings outside the extent of the census data will not be assigned any people.
- [ ] population will not be assigned in zone B (same as above)
- [ ] very unlikely / 'impossible' to have a census area completely within a building. The only possible example i can imagine is a huge multistory residential building. Census boundaries are defined by the number of households ...
- [ ] correct, this methodology is only relevant where the required data exist.

**Comment by Charlotte-Morgan** [comment]
hi @cchristelis - i think we should supply some data for you to work with. I see you have joined population data to RW boundaries. I'm not sure about Indonesian data, but admin boundaries and census boundaries are different in other places.
@timlinux

**Comment by timlinux** [comment]
So a couple of notes from the sprint meeting held on 10 Feb 2015) meeting:

1) 'all buildings in a cencus area should be completely mapped'
2) we will estabilish rules as to what are considered residential and add a new column to the OSM downloader generated dataset to indicate 'residential' based on those rules
3) We should consider people in buildings as a pre-processing tool that can be used to prepare a new type of exposure data as well as an IF that expects this type of exposure as impact layer.

**Comment by davidr007** [comment]
I like the idea of separating the assignment of people in buildings from the IF (i.e. Outcome 3 of your sprint meeting)

Outcome 1 is also important. If you think of people in buildings as the combination of two datasets: (a) the population data; and (b) building data. Outcome 1 ensures that you are not creating  a new dataset in a new location with incomplete data (e.g. buildings).

While the reverse is less likely, shouldn't we also make sure that we do not generate a new dataset where the population data is incomplete (i.e in areas where there are buildings but no population data).

In my opinion the people in buildings pre-processing function should be restricted to areas where both the population and building data are considered complete.

Following this suggestion is safe because of the way you have separated the pre-processing from the IF. In step 1 (pre-processing) you simply place people in buildings everywhere you can. Then, in step 2 (IF) you compute the impact of the hazard on the buildings that have people in them. The analysis region therefore cuts out the 'x' regions and there is no need to extrapolate the census data."

**Comment by timlinux** [comment]
Thanks for the input @davidr007 - a couple of things we should bear in mind:

* how will we know when a cencus area is completely mapped for all the buildings in it
* how do we know is a cencus area is completely mapped of all the people in it (do the Potes data provide such an attribute?)

Agreed about not needing to exatroplate for 'x' if we specify as a basic condition of analysis that data collection should be complete before embarking on analysis for any given area - the onus will be on the user to ensure this is the case. For a starter we will add a disclaimer to that effect in the tool user interface e.g.

`Please note this tool requires that all cencus areas used have complete coverage of buildings, and the cencus data should be complete for the study areas`

One other thought is that we could consider asking HOT to review census areas and tagging each area that does have complete coverage so that we can reliably know which areas have complete surveys.

**Comment by Charlotte-Morgan** [comment]
1. lets look at the metadata on the census data first to understand completeness
2. lets ask the BPS / BNPB what the status of the data are and if there are any holes
3. lets assume the data are complete as they are the 'best available data'
4. lets not make more work ...
5. please remember we are aiming this at local level analysis & the data should be assessed on a case by case basis as it is about to be used.

there is nothing like using data to help the custodians better understand potential uses & limitations

**Comment by cchristelis** [comment]
Thanks for all the input!

I would like you guys to have a look at the alpha version, of the people in buildings tool. It is available on a branch in my repository:
 https://github.com/cchristelis/inasafe/tree/feature/inasafe-1179-people-in-buildings

file:///home/christian/Pictures/Screenshot%20from%202015-02-15%2022:08:13.png

This tool implements the above rules. I am giving a complete procedural breakdown:

*user interaction*
a) The user selects a buildings layer. 
b) The user selects the attribute that holds the levels (in case no levels are stipulated 1 is assumed).
c) The user selects the attribute that holds the usage (Residential is the only type that is considered for population)
d) The user selects a census layer.
e) The user selects the attribute that holds population count.
f) The user may select a output layer, else the new population attribute is added to the buildings layer. (there is a issue at the moment that still causes this to happen)
g) The user hits apply.

*computation*
A) Each census area is evaluated whether it is completely contained within the buildings layer's extents. Those which are not, are ignored.
B) Each building's usage is considered, all non-residential are ignored.
C) The remaining buildings' area overlap with the current census area are computed. 
D) The overlap area of each building is multiplied by levels. Which is summed up to total residential area.
E) The population count is divided by the total residential area of this census giving a population density.
F) This population density is then multiplied by each buildings residential area giving a population estimate per building.
G) The specific population estimate for each building is assigned.

I have been testing this out with some mock data and it seems that the biggest issue we will be facing is data quality...

**Comment by Charlotte-Morgan** [comment]
Hi @cchristelis,

Thanks for the details above & progressing this to a state where we can have a play.
I am unlikely to get a chance to play with the alpha today before our discussion; but in principle it looks good.  I have asked @mahardika to prepare some data for you & I also took a look at some data myself over the weekend because i think we need to get real data to you asap. There will be additional things to consider when you have real data.

Based on the procedure described above - some comments / thoughts:
- [ ] 'residential' for this analysis may not be the basic residential type in OSM data but more likely to be a classification of other standard types. see discussion above
- [ ] the user should be encouraged to click apply gently :)
- [ ] computation a is a problem. We have to assume that the buildings within a census area are all mapped. I would expect the census area could also be larger than the extent of the buildings based on looking at some real data; eg  an island Rokatenga or where there are small villages with clusters of buildings within the census collection area.
- [ ] the other issue with real data is that buildings plot on the boundary of 2 census areas, how will you decide if these are in our out (may I suggest using the building centroid) ?

agree - the biggest issue is data quality. How do you propose doing something sensible if there is not type attribute data on the buildings. can we assume x% are residential ...or will be say the process can't be run?

@davidr007 @mahardika @iyan31 @adissadis

**Comment by cchristelis** [comment]
There is a pull request for creating the exposure layer. This change set is found in:
https://github.com/AIFDR/inasafe/pull/1769
